x/evm/watcher/querier.go

```diff
@@ -80,6 +80,25 @@ func (q Querier) GetBlockByHash(hash common.Hash, fullTx bool) (*EthBlock, error
 	return &block, nil
 }
 
+func (q Querier) GetBlockHashByNumber(number uint64) ([]byte, error) {
+	if !q.enabled() {
+		return nil, errors.New(MsgFunctionDisable)
+	}
+	var height = number
```

Comment:
why not use number directly

Reply:
because there has another condition to change the variable of height
```go
height, err = q.GetLatestBlockNumber()
```